File: server/src/main/java/org/queue/javaapi/producer/SyncProducer.scala
/*
 * Copyright 2010 LinkedIn
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package org.queue.javaapi.producer

import org.queue.javaapi.message.ByteBufferMessageSet
import org.queue.producer.SyncProducerConfig
import org.queue.javaapi.Implicits._
class SyncProducer(syncProducer: org.queue.producer.SyncProducer) {

  def this(config: SyncProducerConfig) = this(new org.queue.producer.SyncProducer(config))

  val underlying = syncProducer

  def send(topic: String, partition: Int, messages: ByteBufferMessageSet) {
    underlying.send(topic, partition, messages)
  }

  def send(topic: String, messages: ByteBufferMessageSet): Unit = send(topic,org.queue.api.ProducerRequest.RandomPartition,
    messages)

  def multiSend(produces: Array[org.queue.javaapi.ProducerRequest]) {

    val produceRequests = new Array[org.queue.api.ProducerRequest](produces.length)
    for(i <- 0 until produces.length)
      produceRequests(i) = new org.queue.api.ProducerRequest(produces(i).topic, produces(i).partition, produces(i).messages)
    underlying.multiSend(produceRequests)
  }

  def close() {
    underlying.close
  }
}

File: server/src/main/java/org/queue/javaapi/producer/SyncProducer.java
package org.queue.javaapi.producer;
import java.nio.ByteBuffer;
import java.util.Arrays;
import org.queue.api.ProducerRequest;
import org.queue.javaapi.ProducerRequest;
import org.queue.message.ByteBufferMessageSet;
import org.queue.producer.SyncProducer;

public class SyncProducer {
    private final SyncProducer underlying;

    public SyncProducer(SyncProducer syncProducer) {
        this.underlying = syncProducer;
    }

    /**
     * Sends the given messages to the specified topic and partition.
     *
     * @param topic      The topic to which messages are sent.
     * @param partition  The partition to which messages are sent.
     * @param messages   The messages to send.
     */
    public void send(String topic, int partition, ByteBufferMessageSet messages) {
        underlying.send(topic, partition, messages);
    }

    /**
     * Sends the given messages to the specified topic using a random partition.
     *
     * @param topic      The topic to which messages are sent.
     * @param messages   The messages to send.
     */
    public void send(String topic, ByteBufferMessageSet messages) {
        send(topic, ProducerRequest.RandomPartition, messages);
    }

    /**
     * Sends multiple messages to various topics and partitions.
     *
     * @param produces   An array of ProducerRequest objects containing topic, partition, and messages.
     */
    public void multiSend(ProducerRequest[] produces) {
        org.queue.api.ProducerRequest[] produceRequests = new org.queue.api.ProducerRequest[produces.length];
        for (int i = 0; i < produces.length; i++) {
            produceRequests[i] = new org.queue.api.ProducerRequest(
                    produces[i].getTopic(),
                    produces[i].getPartition(),
                    produces[i].getMessages()
            );
        }
        underlying.multiSend(produceRequests);
    }

    /**
     * Closes the underlying SyncProducer and releases all resources.
     */
    public void close() {
        underlying.close();
    }
}